src/lib.rs

```diff
@@ -702,13 +702,11 @@ impl<R: gimli::Reader> ResUnit<R> {
         };
 
         if let Some(directory) = file.directory(header) {
-            path_push(
-                &mut path,
-                sections
-                    .attr_string(&self.dw_unit, directory)?
-                    .to_string_lossy()?
-                    .as_ref(),
-            );
+            let directory = sections.attr_string(&self.dw_unit, directory)?;
+            let directory = directory.to_string_lossy()?;
+            if directory != path {
```

Comment:
This doesn't seem exactly right. While unlikely, it would be possible for these to be equal but still need to be relative.

Can we add a special case for directory index 0 instead?

Reply:
sure!